### DISCUSSION TITLE
Configuration to increase deletion rate of Thanos compactor

### DISCUSSION THREAD

**Discussion by anuradhn-coder** [thanos-io/thanos#7632] [Questions & Answers]
Hi,

Our Thanos compactor was not running for almost 7 months (the s3 bucket size too had increased rapidly causing billing issues) and when we checked the logs, it had thrown 'No space left' so we increased the thanos compactor pvc size to 200Gi after which we could see the thanos compactor is working and marking blocks for deletion however the rate at which deletion actually happens is too slow. 

Is there a way to increase deletion rate of Thanos? like to retain only the last one month of data and delete the old previous data before one month

**Comment by anuradhn-coder** [comment]
Hi,

Can anyone help us with this?

**Comment by yeya24** [comment]
Please take a look at https://thanos.io/tip/operating/compactor-backlog.md/ if you haven't read it.
In general, if you have a large amount of backlog, you have to scale up your compactors to catch up.

> Is there a way to increase deletion rate of Thanos? like to retain only the last one month of data and delete the old previous data before one month

This can be done using the retention bucket tool https://thanos.io/tip/operating/compactor-backlog.md/#bucket-tools.
